apps/opik-frontend/src/components/pages/SMEFlowPage/AnnotationView/ThreadDataViewer.tsx

```diff
@@ -71,15 +85,68 @@ const ThreadDataViewer: React.FunctionComponent = () => {
     [tracesData],
   );
 
+  const handleOpenTrace = useCallback(
+    (id: string, shouldFilterToolCalls: boolean) => {
+      // Manage tool filter: add if shouldFilterToolCalls is true, remove if false
+      setTracePanelFilters(
+        manageToolFilter(tracePanelFilters, shouldFilterToolCalls),
+      );
+
+      setTraceId(id);
+      setSpanId("");
+    },
+    [tracePanelFilters, setTracePanelFilters],
+  );
+
+  const handleClose = useCallback(() => {
+    setTraceId("");
+    setSpanId("");
+    // Don't reset filters on close - maintain filter state
+  }, []);
+
+  const handleSetSpanId = useCallback(
+    (
+      updaterOrValue:
+        | string
+        | null
+        | undefined
+        | ((prev: string | null | undefined) => string | null | undefined),
+    ) => {
+      if (typeof updaterOrValue === "function") {
+        setSpanId((prev) => {
+          const newValue = updaterOrValue(prev);
+          return newValue ?? "";
+        });
+      } else {
+        setSpanId(updaterOrValue ?? "");
+      }
+    },
+    [],
+  );
```

Comment:
isn't it worth using 
```
  const [spanId = "", setSpanId] = useQueryParam("span", StringParam, {
    updateType: "replaceIn",
  });
```
instead of this function?

as we have everywhere else in the app

Reply:
Yes, code is short and simpler now ✔️